### PR TITLE
JBIDE-22810 - openshift explorer sometimes shows up empty on windows

### PR DESCRIPTION
**Pull request by robstryker**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by robstryker** [comment]
It's worth noting, the refresh() this patch adds only is called once during createPartControl().  It's not being called during any listeners, events, etc.  So it won't cause tons of refreshes at random times, or lead to the tree expanding and collapsing randomly. 

This patch, as far as I can tell, is 100% effective and has 0 side effects.

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ protected void performOverrides(ILaunchConfigurationWorkingCopy workingCopy) thr
 		String workingDir = JBossServerCorePlugin.getServerStateLocation(s).toOSString();
 		workingCopy.setAttribute(ATTR_WORKING_DIR, workingDir);
     	
-    	Map<String, String> env = workingCopy.getAttribute(ENVIRONMENT_VARS_KEY, (Map)null);
+    	Map<String, String> env = workingCopy.getAttribute(ENVIRONMENT_VARS_KEY, (Map<String,String>)null);
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java]: how is this related and any better?

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/MinishiftLocationSection.java [hunk 2]
```diff
@@ -136,7 +136,7 @@ public void widgetDefaultSelected(SelectionEvent e) {
 		msHomeModListener = new ModifyListener() {
 			@Override
 			public void modifyText(ModifyEvent e) {
-				execute(new SetLocationPropertyCommand(server));
+				execute(new SetMinishiftHomePropertyCommand(server));
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/MinishiftLocationSection.java]: how is this minishift/cdk change related to the original jira?

File: plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/OpenShiftExplorerView.java [hunk 3]
```diff
@@ -115,6 +115,7 @@ public void createPartControl(Composite parent) {
 		this.explanationsPane = createExplanationPane(connectionsPane, pageBook, toolkit);
 		showConnectionsOrExplanations(connectionsPane, explanationsPane);
 		ConnectionsRegistrySingleton.getInstance().addListener(this);
+		getCommonViewer().refresh();
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/OpenShiftExplorerView.java]: this seems like the only valid change here. Is it? does it fix the issue?

**Comment by adietish** [comment]
@robstryker you also need to add the "signed-off" to your commit message (see failing check above)

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java (hunk 1 above)
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java]: The code had a warning icon. I cleaned it up.

File: plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/OpenShiftExplorerView.java (hunk 3 above)
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/OpenShiftExplorerView.java]: This fixes the main issue.

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/MinishiftLocationSection.java (hunk 2 above)
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/MinishiftLocationSection.java]: This fixes JBIDE-24796.   Sorry for the confusion, I was using this branch as a collector until master is open.

Comment (adietish) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/MinishiftLocationSection.java]: dare to untangle the patches then, please?
